web/onos-gui/src/app/onos-topo/device-detail/device-detail.component.html

```diff
@@ -76,49 +68,78 @@ <h2 class="editable clickable">{{detailsData?.getId()}}</h2>
+                    </tr>
+                    <tr>
+                        <td class="label" width="110">Password :</td>
+                        <td class="value" width="80">{{detailsData?.getCredentials().getPassword()}}</td>
+                    </tr>
                 </tbody>
             </table>
             <h2 class="changes-title">TLS Parameters</h2>
             <table>
                 <tbody>
-                <tr>
-                    <td class="label" width="110">Cert Authority :</td>
-                    <td class="value"
-                        width="80">{{detailsData?.getTls().getCacert()}}</td>
-                </tr>
-                <tr>
-                    <td class="label" width="110">Client Cert :</td>
-                    <td class="value"
-                        width="80">{{detailsData?.getTls().getCert()}}</td>
-                </tr>
-                <tr>
-                    <td class="label" width="110">Client Key:</td>
-                    <td class="value"
-                        width="80">{{detailsData?.getTls().getKey()}}</td>
-                </tr>
-                <tr>
-                    <td class="label" width="110">Plain connection :</td>
-                    <td class="value"
-                        width="80">{{detailsData?.getTls().getPlain()}}</td>
-                </tr>
-                <tr>
-                    <td class="label" width="110">Insecure connection :</td>
-                    <td class="value"
-                        width="80">{{detailsData?.getTls().getInsecure()}}</td>
-                </tr>
-
+                    <tr>
+                        <td class="label" width="110">Cert Authority :</td>
+                        <td class="value" width="80">{{detailsData?.getTls().getCacert()}}</td>
+                    </tr>
+                    <tr>
+                        <td class="label" width="110">Client Cert :</td>
+                        <td class="value" width="80">{{detailsData?.getTls().getCert()}}</td>
+                    </tr>
+                    <tr>
+                        <td class="label" width="110">Client Key:</td>
+                        <td class="value" width="80">{{detailsData?.getTls().getKey()}}</td>
+                    </tr>
+                    <tr>
+                        <td class="label" width="110">Plain connection :</td>
+                        <td class="value" width="80">{{detailsData?.getTls().getPlain()}}</td>
+                    </tr>
+                    <tr>
+                        <td class="label" width="110">Insecure connection :</td>
+                        <td class="value" width="80">{{detailsData?.getTls().getInsecure()}}</td>
+                    </tr>
+                </tbody>
+            </table>
+            <h2 class="changes-title">Attributes</h2>
+            <table>
+                <tbody>
+                    <tr>
+                        <td class="label" width="110">Longitude :</td>
```

Comment:
I should have said - this needs to be dynamically driven - the attributes could have anything in them at run time, so this needs a ngFor loop and "| keyvalue"